### PR TITLE
Add hacky support for injecting into stream into Next.js

### PR DESCRIPTION
**Pull request by preslavle**
Make stream work!

### CODE REVIEW

File: packages/next/src/server/app-render.tsx [hunk 1]
```diff
@@ -2026,7 +2031,19 @@ export async function renderToHTMLOrFlight(
...
+            const div = `<div id="${escapeQuote(
+              obj.id
+            )}" data-value="${escapeQuote(
+              obj.value
+            )}">Michal is the greatest</div>`
+            console.log(`Injecting into stream: ${div}`)
+            return div
+          })
+          .join('')
+        return result.pipeThrough(createSuffixStream(injected))
```
Comment (xixixao) [inline packages/next/src/server/app-render.tsx]: Moving this out will make this quite painless!
```suggestion
        const injected = nextInjectToStreamBuffer.join('')
        return result.pipeThrough(createSuffixStream(injected))
```

**Review by xixixao — APPROVED** [review]
Let's see if navigation works and if we can deploy this to Vercel.

File: packages/next/src/server/node-web-streams-helper.ts [hunk 2]
```diff
@@ -332,13 +332,11 @@ let nextInjectToStreamBuffer: string[] = []
 export function customInjectStream(): TransformStream<Uint8Array, Uint8Array> {
   return new TransformStream({
     async transform(chunk, controller) {
+      controller.enqueue(chunk)
```
Comment (sujayakar) [inline packages/next/src/server/node-web-streams-helper.ts:336]: nice, this guarantees our chunk comes before the rendered component chunks?

Comment (preslavle) [reply packages/next/src/server/node-web-streams-helper.ts:336]: I think they are sent after but are sent in the same HTML batch. Not 100% sure but seem good enough for our purposes.

**Pull request merged**